### PR TITLE
Add imagenet_cartoon and imagenet_drawing

### PR DESCRIPTION
**Pull request by tiagosalvador**
# Task Description
Two datasets for benchmarking machine learning robustness against domain shifts, based on the ImageNet dataset and using using a GAN framework and simple image processing.

examples
Several examples of ImageNet images (top) and their respective ImageNet-Cartoon (middle) and ImageNetDrawing (bottom) versions.

## Dataset Creation
Images are taken from the ImageNet dataset and the transformed into cartoon using the GAN framework proposed by [1] and into drawing using the simple image processing described in [2].

[1] Wang, X. and Yu, J. Learning to cartoonize using white-box cartoon representations. In _Proceedings of the IEEE/CVF
Conference on Computer Vision and Pattern Recognition (CVPR)_, June 2020.
[2] Lu, C., Xu, L., and Jia, J. Combining Sketch and Tone for Pencil Drawing Production. In Asente, P. and Grimm, C.
(eds.), _International Symposium on Non-Photorealistic Animation and Rendering_. The Eurographics Association, 2012. ISBN 978-3-905673-90-6. doi: 10.2312/PE/NPAR/NPAR12/065-073.

## Evaluation Metrics
Robust accuracy: correct classification of the transformed images.

## Expected Insights/Relevance
The accuracy of pretrained ImageNet models decreases significantly on the proposed datasets.

## Access
We release the dataset on Zenodo (https://zenodo.org/record/6801109), as well as the code to generate the two datasets  (https://github.com/oberman-lab/imagenet-shift).

## Data
The dataset is hosted on Zenodo (https://zenodo.org/record/6801109).

## License
We released the data with the [Creative Commons Attribution 4.0 International](https://creativecommons.org/licenses/by/4.0/legalcode).

### CODE REVIEW

**Review by stes — CHANGES_REQUESTED** [review]
Hi @tiagosalvador , thanks a lot for your contribution. We are working towards the first release of the workshop and for this, released our [contribution guide](https://github.com/shift-happens-benchmark/icml-2022/blob/main/CONTRIBUTING.rst) with the aim of unifying the implementation of all contributions.

Please have a look, the required changes should be relatively small and are mostly concerns with placing the files within the package, and adding documentation.

From reviewing your PR, the main required changes are:

- [ ] Place the files directly in the package instead of the example order
- [ ] Add a README and optionally citation info
- [ ] Add more description to your python files (to the docstring -- the documentation will be auto generated from this)

Thanks a lot for your contribution, and please ping me if you have any questions. If you have technical questions, also feel free to ping @kotekjedi who is happy to help with implementation details.

**Comment by stes** [comment]
Hi!

This is an automated report for the documentation coverage of your PR. To provide good user experience for the benchmark, we aim to cover most functions with doc strings, using the [Google docstring format](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html).

We would greatly appreciate your help in assuring that docstring coverage is sufficiently high for your contribution.

The current coverage is:
| Name                     |      Total |      Miss |      Cover |      Cover% |
|--------------------------|------------|-----------|------------|-------------|
| imagenet_cartoon.py      |          3 |         2 |          1 |         33% |
| imagenet_drawing.py      |          3 |         2 |          1 |         33% |
|--------------------------|------------|-----------|------------|-------------|
| TOTAL                    |          6 |         4 |          2 |       33.3% |
RESULT: FAILED (minimum: 80.0%, actual: 33.3%)

You can reproduce this result by running

```
interrogate -v --ignore-nested-functions --ignore-private --ignore-semiprivate --ignore-init-method
```

on the files you added to your PR.

**Review by zimmerrol — CHANGES_REQUESTED** [review]
Thanks for this PR! Once the examples/* files are removed (since you moved them, as requested, to the shifthappens/tasks package), we should be good to merge this PR.

File: shifthappens/tasks/imagenet_cartoon/__init__.py [hunk 1]
```diff
@@ -20,20 +20,31 @@
...
 @dataclasses.dataclass
-class ImageNetR(Task):
+class ImageNetCartoon(Task):
+    """ImageNet-Cartoon Dataset.
+    This task evaluates a model on ImageNet-Cartoon. This 
+    dataset was formed by converting the images in the
+    ImageNet validation set into cartoons using a GAN
+    framework. See the readme file for more information
+    about how the dataset was constructed.
+
```
Comment (kotekjedi) [inline shifthappens/tasks/imagenet_cartoon/__init__.py:33]:
```suggestion

```

File: shifthappens/tasks/imagenet_cartoon/__init__.py [hunk 2]
```diff
@@ -20,20 +20,31 @@
...
+    """ImageNet-Cartoon Dataset.
+    This task evaluates a model on ImageNet-Cartoon. This 
+    dataset was formed by converting the images in the
+    ImageNet validation set into cartoons using a GAN
+    framework. See the readme file for more information
+    about how the dataset was constructed.
+    
+    The goal of this evaluation task is to measure the
+    model's robustness to distribution shifts.
+    """
```
Comment (kotekjedi) [inline shifthappens/tasks/imagenet_cartoon/__init__.py:36]:
```suggestion
    """
    
```

**Comment by kotekjedi** [comment]
Hi Tiago!
My name is Alexander Panfilov; I am a student research assistant working on supporting the project. I reviewed the PR and left some comments - two are about paths, and the other is to pass black formatter.

File: shifthappens/tasks/imagenet_drawing/__init__.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    ]
+
+    def setup(self):
+        """Setup ImageNet-Drawing"""
+        dataset_folder = os.path.join(self.data_root, "imagenet-drawing")
+        if not os.path.exists(dataset_folder):
+            # download data
+            for file_name, url, md5 in self.resources:
+                sh_utils.download_and_extract_archive(
+                    url, self.data_root, md5, file_name
```
[the remark this answers is not in the document]
Comment (tiagosalvador) [reply shifthappens/tasks/imagenet_drawing/__init__.py]: Yes, it should.

File: shifthappens/tasks/imagenet_cartoon/__init__.py [hunk 4]
```diff
@@ -20,20 +20,31 @@
...
         )
     ]
 
     def setup(self):
-        dataset_folder = os.path.join(self.data_root, "imagenet-r")
+        """Setup ImageNet-Cartoon"""
+        dataset_folder = os.path.join(self.data_root, "imagenet-cartoon")
         if not os.path.exists(dataset_folder):
             # download data
             for file_name, url, md5 in self.resources:
```
[the remark this answers is not in the document]
Comment (tiagosalvador) [reply shifthappens/tasks/imagenet_cartoon/__init__.py:50]: Yes, it should.

**Comment by tiagosalvador** [comment]
Hi Alexander! I made the necessary changes. Thanks for reviewing it!

**Pull request merged**